src/Illuminate/Collections/Collection.php

```diff
@@ -793,10 +793,16 @@ public function only($keys)
      */
     public function pop($count = 1)
     {
+        if ($count < 1 || $this->count() === 0) {
+            return null;
```

Comment:
This should probably be a new empty collection instead of null to allow for chaining?

Reply:
I think you can just use `return;`, there are 2 more spots in this file related to the functions `median()` and `mode()` which also have a null return. I'm not sure if returning a new collection is appropriate here when comparing it to the php `array_pop()`

Reply:
as mentioned by @bert-w with php the functions `array_pop` and `array_shift` returning `null ` in the case of an empty array, when the $count parameter is `1` this implementation seems appropriate.

However, in the case that there is a user supplied parameter which is not `1` things get a little bit more difficult...
Considering the current behaviour of the function it may be better to apply something similar to the following logic

```
if ($count === 1) {
    return object | null;
}

if ($count < 0 || $this->count() === 0) {
    return empty collection;
}

return $count item collection;
```

The more I think about it, it is probally more appropriate to seperate the funtionality into classic `pop` functionality returning `object | null`, and `popMultiple`  which takes a parameter and returns a collection. (As this is already in production it would be a breaking change?)